lib/Runtime/Base/FunctionBody.h

```diff
@@ -1765,60 +1755,60 @@ namespace Js
     protected:
         static HRESULT MapDeferredReparseError(HRESULT& hrParse, const CompileScriptException& se);
 
-        Field(bool) m_hasBeenParsed : 1;       // Has function body been parsed- true for actual function bodies, false for deferparse
-        Field(bool) m_isDeclaration : 1;
-        Field(bool) m_isAccessor : 1;          // Function is a property getter or setter
-        Field(bool) m_isStaticNameFunction : 1;
-        Field(bool) m_isNamedFunctionExpression : 1;
-        Field(bool) m_isNameIdentifierRef  : 1;
-        Field(bool) m_isClassMember : 1;
-        Field(bool) m_isStrictMode : 1;
-        Field(bool) m_isAsmjsMode : 1;
-        Field(bool) m_isAsmJsFunction : 1;
-        Field(bool) m_isWasmFunction : 1;
-        Field(bool) m_isGlobalFunc : 1;
-        Field(bool) m_doBackendArgumentsOptimization : 1;
-        Field(bool) m_doScopeObjectCreation : 1;
-        Field(bool) m_usesArgumentsObject : 1;
-        Field(bool) m_isEval : 1;              // Source code is in 'eval'
-        Field(bool) m_isDynamicFunction : 1;   // Source code is in 'Function'
-        Field(bool) m_hasImplicitArgIns : 1;
-        Field(bool) m_dontInline : 1;            // Used by the JIT's inliner
+        FieldWithBarrier(bool) m_hasBeenParsed : 1;       // Has function body been parsed- true for actual function bodies, false for deferparse
```

Comment:
also I think we should make FieldWithBarrier to be force write barrier, regardless of the type, to be able to use write barrier on non-pointer field like intptr_t, there's no such use case for now though.

---
In reply to: [90977538](https://github.com/Microsoft/ChakraCore/pull/2144#discussion_r90977538) [](ancestors = 90977538,90976930)